specs/schemes/exact/scheme_exact_algo.md

```diff
@@ -233,23 +120,154 @@ Algorand uses [**msgpack**](https://www.npmjs.com/package/algorand-msgpack) to e
-  3.1 `PaymentTxID` is supplied in the payment HEAD, but can and **SHOULD** be calculated by hashing the encoded transaction parameters with SHA512_256 and encoding the result in base32 with a length of 52 characters.
+1. Check the `paymentGroup` contains 16 or fewer elements.
+1. Decode all transactions from the `paymentGrouop`.
+1. Locate the `paymentGroup[paymentIndex]` transaction from the `Payment Payload`.
+    1. Check the `aamt` (asset amount) matches `maxAmountRequired` from the `Payment Requirements`.
+    1. Check the `arcv` (asset receiver) matches `payTo` from the `Payment Requirements`.
+1. Locate all transactions where for `snd` (sender) is the `Facilitator`s Algorand address.
+    1. Check the `type` (transaction type) is `pay`.
+    1. Check the following fields are omitted: `close`, `rekey`, `amt`.
+    1. Check the `fee` (Fee) is a reasonable amount.
+    1. Sign the transaction.
+1. Evaluate the payment group against an Algorand node's `simulate` endpoint to ensure the transactions would succeed.
 
 ## Settlement
 
-Once the group is validated by the resource server, settlement can occur by the facilitator submitting the `paymentGroup` to the Algorand network through the `v2/transactions` endpoint against any valid Algorand node.
+Once the group is validated by the resource server, settlement can occur by the facilitator submitting the verified transaction group to the Algorand network through the `v2/transactions` endpoint against any valid Algorand node.
 
-In Algorand there are no consensus forks and so it achieves finality instantly a transaction is included in a block. So as soon as the transaction is included in a block, the payment is considered settled and the facilitator can inform the resource server of the successful payment and proceed with the resource delivery.
+In Algorand there are no consensus forks and so it achieves instant finality the moment a transaction is included in a block. So as soon as the transaction is included in a block, the payment is considered settled and the facilitator can inform the resource server of the successful payment and proceed with the resource delivery.
 
 ## Additional Considerations
 
-- In order for the resource server to receive payment on a particular Asset, it **MUST** be opted-in to that asset.
+### Assets
+
+In order for the resource server to receive payment on a particular Asset, it **MUST** be opted-in to that asset. In Algorand, accounts explicitly enable receiving a particular asset, otherwise an asset transfer to that account will be rejected. So as part of the resource server's setup, it **SHOULD** ensure it has opted-in to the asset ID specified in the `paymentRequirements.asset`.
```

Comment:
Do you think we should mention something about the clawback here or is too much detail?

Reply:
Agreed in slack to keep this specific clawback scenario out of the spec, but at least it's possible.